Tests/lib/vso-task-lib/lib/vsotask.ts

```diff
@@ -91,7 +91,9 @@ export function exit(code: number): void {
 export function getVariable(name: string): string {
     var varval = process.env[name.replace(/\./g, '_').toUpperCase()];
     debug(name + '=' + varval);
-    return varval;
+
+    var mocked =  mock.getResponse('getVariable', name);
+    return varval || mocked;
```

Comment:
the mocked value should take precedence so your can assert stuff reliably in your tests.

```
return mocked || varval
```